tools/refenginediscovery/refenginediscovery.go

```diff
@@ -37,14 +37,14 @@ func Discover(ctx context.Context, protocols []string, host string, refEngineCal
 		protocols = []string{"https", "http"}
 	}
 
-	uri, err := url.Parse("https://example.com/.well-known/oci-host-ref-engines")
+	uri, err := url.Parse("https://" + host + "/.well-known/oci-host-ref-engines")
```

Comment:
The previous approach avoided this unnecessary string concatenation.  And once we add DNS-ancestor walking, we'll be setting `uri.Host` anyway.